packages/api/endpoints/s3credentials.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const router = require('express-promise-router')();
+
+const { lambda } = require('@cumulus/common/aws');
+
+/**
+ * Use NGAP's time-based, temporary credential dispensing lambda.
+ *
+ * @param {string} username - earthdata login username
+ * @returns {Promise<Object>} Payload containing AWS STS credential object valid for 1
+ *                   hour.  The credential object contains keys: AccessKeyId,
+ *                   SecretAccessKey, SessionToken, Expiration and can be use
+ *                   for same-region s3 direct access.
+ */
+async function requestTemporaryCredentialsFromNgap(username) {
+  const FunctionName = 'gsfc-ngap-sh-s3-sts-get-keys';
```

Comment:
Should this be an environment variable or something?

Reply:
Good question, I had a comment above it before asking if we should be passing this in via ENVVAR.

I don't know the answer, I think it is probably wrapped up in what should this do for non-NGAP deploys.  NGAP has committed to supporting this lambda for users, I don't see that changing.  I suppose the only real value in making it a envvar would be to be able to update it on the lambda console without a redeploy.  It def made my eye twitch, but I don't know if adding it as an ENVVAR buys that much (it's particularly tied to this function)

Reply:
I'm fine with it either way. I don't think it buys us a ton. This is a good point, though, and I think we need to call out in the API documentation (and any other documentation we end up writing) that this functionality is specific to NGAP.

Reply:
@markdboyd talked with Lauren, I will update this with the option to use an environment variable and document that in non-ngap environments you can set that variable to a function that takes a username and returns time based security tokens via the AWS STS service.   But that it will default to the currently hard-coded function name for the NGAP dispenser.  

This seems like it will work fine, and if a non-ngap doesn't need credentials, you don't have to implement this functionality.  Seem reasonable?

Reply:
a146eaa